docs/source/developing/tasks.rst

```diff
@@ -0,0 +1,92 @@
+Adding Tasks to the pyAFQ Workflow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+pyAFQ provides methods for performing tractography, registration, bundle
+recognition, visualization, tract profiling, and other elements of tractometry.
+Additionally, pyAFQ provides a simple UI (which can be accessed as an API or
+CLI) for running part or all of the tractometry pipeline on a given dataset.
+
+This UI uses `Pimms <http://bids.neuroimaging.io/>`_ (a Python immutable
+data structures library) as its workflow engine. Individual tasks in this workflow
+are specified in the AFQ.tasks folder. The tasks are split up into 5 files:
+data, mapping, tractography, segmentation, and visualization (viz). If you want
+to add new functionality to the pyAFQ workflow, first decide which file to
+put it in. Try to keep similar functionality in the same file. All tasks are
+decorated with `@pimms.calc("output_name")`, where `output_name` is used as
+input to other tasks and can be accessed by the user through the
+:class:`AFQ.api.AFQ` object.
+
+When a user instantiates an :class:`AFQ.api.AFQ` object, a Pimms plan is
+created using a selection of tasks based on the configuration provided by
+the user. Each task file has a `get_{task filename}_plan` method which is
+called by the :class:`AFQ.api.AFQ` class (for example, in AFQ.tasks.data
+there is: :func:`AFQ.tasks.data.get_data_plan`). This method uses the user's
+configuration choices (which it takes as input) to construct a Pimms plan
+from the tasks in the file. In most cases, new tasks can simply be added to
+the list of tasks being passed to `with_name`. These tasks are always added
+to the plan; however they will only be called if the user needs them. In a few
+cases, additional logic is required, like when two different tasks compute the
+same output, or if the task is from an AFQ.definition.
```

Comment:
I don't understand the latter part of this sentence. Does this mean something like "if the task is inherited from `AFQ.Definition`"?

Reply:
I clarified it:  "or if the task being provided by a `Definition`-inherited object
(see the source of :func:`AFQ.tasks.mapping.get_mapping_plan` for an example
of both)."